sql/analyzer/autocommit_test.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 DoltHub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analyzer
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/parse"
+
+	"github.com/stretchr/testify/require"
+)
+
+var testCasesWithoutCurrentDb = map[string]string{
+	"CREATE PROCEDURE mydb.p1() SELECT 5": "mydb",
+	"create database db123":               "db123",
+}
+
+var testCasesWithCurrentDb = map[string]string{
+	"use db1":                                "db1",
+	"select 1":                               "foo",
+	"select * from t1":                       "foo",
+	"call myFunction()":                      "foo",
+	"insert into db2.t2 values(1, 2, 3)":     "db2",
+	"insert into t2 values(42, 42)":          "foo",
+	"create table db1.t3(i int primary key)": "db1",
+	"create database db123":                  "db123",
+}
+
+var multiDbErrorTestCases = []string{
```

Comment:
I'm not comfortable giving up this functionality.

We need the ability to select and join across multiple DBs in a query, and selects are harmless to this end.

What we want to catch and error on are modifications of two or more DBs in the same transaction, and prevent it.

Longer term we need to redefine transactions as belonging to either a `Session` or `DatabaseProvider`, rather than a `Database`, and let the integrator decide when it's appropriate to do a write to more than one DB atomically. For now the answer is "never" and we need to enforce that.